pkl-core/src/main/java/org/pkl/core/project/CanonicalPackageUri.java

```diff
@@ -71,17 +67,12 @@ public static CanonicalPackageUri of(String uriStr) throws URISyntaxException {
     return new CanonicalPackageUri(baseUri, majorVersion);
   }
 
-  public CanonicalPackageUri(URI baseUri, int majorVersion) {
-    this.baseUri = baseUri;
-    this.majorVersion = majorVersion;
-  }
-
-  @SuppressWarnings("unused")
+  @Deprecated
```

Comment:
For each of these: add `forRemoval = true`, and add a doc comment about what to replace with

```suggestion
  /**
   * @deprecated As of 0.28.0, replaced by {@link #majorVersion()}.
   */
  @Deprecated(forRemoval = true)
```